### PR TITLE
Hardening/two raised to

### LINKED ISSUE #96: move two_raised_to() into fast_math component and test

### PR DESCRIPTION
**Pull request by adam-mcdaniel**
Moved `two_raised_to` to `fm` namespace in `fast_math` component and added a test.

closes #96

### CODE REVIEW

**Comment by grahamlopez** [comment]
okay to test

**Comment by grahamlopez** [comment]
@adam-mcdaniel was there an issue for this?

**Comment by grahamlopez** [comment]
test this please

**Comment by grahamlopez** [comment]
> @adam-mcdaniel was there an issue for this?

Nevermind, I see it now #96

File: src/fast_math.hpp [hunk 1]
```diff
@@ -2,8 +2,13 @@
 #include "lib_dispatch.hpp"
 #include "tensors.hpp"
 
+// // a non-matlab one-liner that had no better home - compute 2^arg
+// inline int two_raised_to(int exponent) { return 1 << exponent; }
```
Comment (grahamlopez) [inline src/fast_math.hpp]: we don't need these two commented-out lines

File: src/fast_math.hpp [hunk 2]
```diff
@@ -2,8 +2,13 @@
...
+
 namespace fm
 {
+// a non-matlab one-liner that had no better home - compute 2^arg
+inline int two_raised_to(int exponent) { return 1 << exponent; }
```
Comment (grahamlopez) [inline src/fast_math.hpp]: add an extra blank line below here to separate from `axpy()`

Comment (grahamlopez) [inline src/fast_math.hpp]: we need to protect against negative inputs for `exponent`

File: src/fast_math_tests.cpp [hunk 3]
```diff
@@ -1,6 +1,18 @@
 #include "fast_math.hpp"
 #include "tensors.hpp"
 #include "tests_general.hpp"
+#include <cmath>
+
+TEST_CASE("fm::two_raised_to", "[fast_math]")
+{
+  SECTION("pow")
+  {
+    for (int i = 1; i < 31; i++)
```
Comment (grahamlopez) [inline src/fast_math_tests.cpp]: why not test '0'?

Comment (adam-mcdaniel) [reply src/fast_math_tests.cpp]: I didn't consider it, I'll add it in

File: src/fast_math.hpp (hunk 2 above)
Comment (adam-mcdaniel) [reply src/fast_math.hpp]: Would `unsigned int` work or would an assert be better?

File: src/fast_math.hpp (hunk 1 above)
Comment (adam-mcdaniel) [reply src/fast_math.hpp]: Oops

File: src/fast_math.hpp (hunk 2 above)
Comment (adam-mcdaniel) [reply src/fast_math.hpp]: I put an assert in the newest commit

File: src/fast_math_tests.cpp (hunk 3 above)
Comment (adam-mcdaniel) [reply src/fast_math_tests.cpp]: Fixed with newest commit

File: src/fast_math.hpp (hunk 2 above)
Comment (grahamlopez) [reply src/fast_math.hpp]: No to unsigned. For your own reference - advice from the committee: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-nonnegative

**Comment by grahamlopez** [comment]
Nice job. Merging.

**Pull request merged**